state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1682,14 +1687,48 @@ pub mod pallet {
 			Ok(())
 		}
 
-		fn check_swap_price_violation(swap: &SwapState<T>) -> Result<(), SwapFailureReason> {
+		fn check_swap_price_violation(
+			swap: &SwapState<T>,
+		) -> Result<Option<SignedBasisPoints>, SwapFailureReason> {
+			let input_oracle_price = T::PriceFeedApi::get_price(swap.input_asset());
+			let output_oracle_price = T::PriceFeedApi::get_price(swap.output_asset());
+
+			// Calculate executed price delta from the oracle price.
+			let oracle_delta = if let (Some(input_oracle), Some(output_oracle)) =
```

Comment:
As discussed, we could put this in a separate function considering that this delta is not used in the actual checks below (but it almost looks like it might).